document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
```

Comment:
I'd like to leave this as is. The requirement for tools to parse logs makes it useful for the id to be stable. And "opaque" doesn't mean "unreadable by a human", it means "unparsable by a machine" (that is, a machine should not assume that the identifier has any particular internal structure or syntax).

---
In reply to: [61357910](https://github.com/sarif-standard/sarif-spec/pull/172#discussion_r61357910) [](ancestors = 61357910)